lib/livebook_web/live/user_component.ex

```diff
@@ -85,27 +88,38 @@ defmodule LivebookWeb.UserComponent do
 
   @impl true
   def handle_event("randomize_color", %{}, socket) do
-    data = %{
-      socket.assigns.data
-      | "hex_color" => User.random_hex_color(except: [socket.assigns.preview_user.hex_color])
-    }
-
-    handle_event("validate", %{"data" => data}, socket)
+    hex_color = User.random_hex_color(except: [socket.assigns.preview_user.hex_color])
+    handle_event("validate", %{"user" => %{"hex_color" => hex_color}}, socket)
   end
 
-  def handle_event("validate", %{"data" => data}, socket) do
-    {valid, user} =
-      case User.change(socket.assigns.user, data) do
-        {:ok, user} -> {true, user}
-        {:error, _errors, user} -> {false, user}
+  def handle_event("validate", %{"user" => attrs}, socket) do
+    params = Map.merge(socket.assigns.changeset.params, attrs)
```

Comment:
I don't think we need to merge this because `attrs` should have the previous params?

Reply:
If the color is the issue, then you should always store the color in a hidden form field, so we always get it as a parameter. :)

Reply:
Yeah, I don't need in this case